src/pages/home-page/HomePage.jsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect } from "react";
+import axios from "axios";
+import "./home-page.css";
+import { Category, Header, useDataFromServer } from "../../components";
+import { tshirt } from "../../assets/images";
+
+const HomePage = () => {
+  const { state, dispatch } = useDataFromServer();
+
+  const fetchData = async () => {
```

Comment:
A suggestion, you can move this block inside the useEffect callback just before calling the function.

Reply:
I will make it as immediately invoked function.